packages/xrpl/test/integration/transactions/oracleSet.test.ts

```diff
@@ -0,0 +1,84 @@
+import { assert } from 'chai'
+
+import { OracleSet } from '../../../src'
+import serverUrl from '../serverUrl'
+import {
+  setupClient,
+  teardownClient,
+  type XrplIntegrationTestContext,
+} from '../setup'
+import { testTransaction } from '../utils'
+
+// how long before each test case times out
+const TIMEOUT = 20000
+
+describe('OracleSet', function () {
+  let testContext: XrplIntegrationTestContext
+
+  beforeEach(async () => {
+    testContext = await setupClient(serverUrl)
+  })
+  afterEach(async () => teardownClient(testContext))
+
+  it(
+    'base',
+    async () => {
+      const tx: OracleSet = {
+        TransactionType: 'OracleSet',
+        Account: testContext.wallet.classicAddress,
+        OracleDocumentID: 1234,
+        LastUpdateTime: 4294967295, // Date.now() - 946684800,
+        // Provider: '70726F7669646572',
+        // URI: '6469645F6578616D706C65',
+        // AssetClass: 'currency',
+        // PriceDataSeries: [
+        //   {
+        //     BaseAsset: 'XRP',
+        //     QuoteAsset: 'USD',
+        //     AssetPrice: 740,
+        //     Scale: 3,
+        //   },
+        // ],
```

Comment:
Why is this commented out?

Reply:
It now fails with this error:

```
 FAIL   xrpl.js  test/integration/transactions/oracleSet.test.ts
  ● OracleSet › base

    Cannot read FieldOrdinal, type_code out of range

      145 |       type = this.readUInt8()
      146 |       if (type === 0 || type < 16) {
    > 147 |         throw new Error('Cannot read FieldOrdinal, type_code out of range')
          |               ^
      148 |       }
      149 |     }
      150 |

      at BinaryParser.readFieldOrdinal (../ripple-binary-codec/src/serdes/binary-parser.ts:147:15)
      at BinaryParser.readFieldOrdinal [as readField] (../ripple-binary-codec/src/serdes/binary-parser.ts:167:51)
      at Function.readField [as fromParser] (../ripple-binary-codec/src/types/st-object.ts:71:28)
      at STArray.fromParser [as toJSON] (../ripple-binary-codec/src/types/st-array.ts:99:36)
      at STObject.toJSON (../ripple-binary-codec/src/types/st-object.ts:185:10)
      at toJSON (../ripple-binary-codec/src/binary.ts:47:53)
      at readJSON (../ripple-binary-codec/src/binary.ts:60:18)
      at binaryToJSON (../ripple-binary-codec/src/index.ts:33:10)
      at hashSignedTx (src/utils/hashes/hashLedger.ts:77:22)
      at Wallet.sign (src/Wallet/index.ts:436:25)
      at sign (src/sugar/submit.ts:258:17)
      at fulfilled (src/sugar/submit.ts:2354:24)
```

Reply:
Oh, looks like `STCurrency` is a new field type introduced in Price Oracles amendment that's not in xrpl.js; this type is for `BaseAsset` and `QuoteAsset`. That's probably why it fails with `FieldOrdinal` error?

Reply:
The type exists, though: https://github.com/XRPLF/xrpl.js/blob/main/packages/ripple-binary-codec/src/types/currency.ts

Reply:
I'd recommend adding a binary codec test case; it'll be easier to debug that way.